src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -721,6 +725,29 @@ public override void CloseAdditionalDocument(DocumentId documentId)
             CloseDocumentCore(documentId);
         }
 
+        public bool TryOpenFile(string filePath, ProjectId projectId = null, bool preview = false)
```

Comment:
sure I can move it out. I just wanted to share some method in there such as TryGetFrame and etc. I can move all those out.